client/question/services/code_preprocessors/PythonCodePreprocessorServiceSpec.js

```diff
@@ -34,6 +37,28 @@ describe('PythonCodePreprocessorService', function() {
       'PerformanceTestObjectFactory');
   }));
 
+  describe('_prepareCodeSubmissionForServerExecution', function() {
+
+    it('appends the correct code to the supplied code', function() {
+      var code = [
+        'def myFunction(arg):',
+        '    result = arg.rstrip()',
+        '    return result',
+        ''
+      ].join('\n');
+      var codeSubmission = CodeSubmissionObjectFactory.create(code);
+      PythonCodePreprocessorService._prepareCodeSubmissionForServerExecution(
+          codeSubmission);
+      expect(codeSubmission.getPreprocessedCode()).toContain(
```

Comment:
I think you should just assert that the entire block of preprocessed code matches the hardcoded expected block of code. You want to also ensure that the student's code also appears in the right location and doesn't get mangled.